### ISSUE TITLE
Auto-complete behavior with quotes doesn't work correctly when trying to add quotes to an existing string

### ISSUE THREAD

**Issue by theficus** [adamdriscoll/poshtools#230]
Consider this scenario:

```powershell
Write-Host hello world
```
If I want to surround "hello world" in quotes, when I type a quotation mark at "hello", it inserts two quotation marks like this:
```powershell
Write-Host ""hello world
```
Then I need to go back and fix it manually.

This doesn't seem like correct behavior.

**Comment by theficus** [comment]
This same issue also happens with parenthesis, single ticks, and curly braces. It would appear that the auto-completion is a bit too aggressive.

**Comment by bmoore-msft** [comment]
yep, that only makes sense on the when you're on the EOL and need completion (i.e. not if you already have the first char of the pair)

**Comment by AndreSayreMSFT** [comment]
Agreed.

**Comment by AndreSayreMSFT** [comment]
Is this related to Microsoft#532 ?

**Comment by AndreSayreMSFT** [comment]
We haven't yet gotten to a complete solution for the full issue:

eg, :
write-host hello|
write-host "|hello
[type "at EOL, as we still have an open, unpaired "]
write-host "hello""|

expected: write-host "hello"|
